### ISSUE TITLE
feat: Unit test execution for rust tests

### ISSUE THREAD

**Issue by sauravpanda** [Cloud-Code-AI/kaizen#379]
We need to add functionality so users can use Kaizen to run unit tests and show the results. This will also help our agentic loop with actual execution feedback so that we can fix the test cases.

**Comment by sauravpanda** [comment]
@MashyBasker were you trying to set the docker up for unit test?

btw would need your operating system details, seems like a mismatch is there!

**Comment by MashyBasker** [comment]
Yes. I created a Github app, adding the Home URL and Webhook(HookDeck), configured the permissions and tried to deploy the API using the `GITHUB_APP_KEY.pem` file for authentication.

Here are my OS details:

```
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
VERSION="6.9.5-arch1-1"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo
```
@sauravpanda

**Comment by sauravpanda** [comment]
I think for this, you dont need GitHub so it should be good, right?

**Comment by MashyBasker** [comment]
Yeah. Do you think it might be caused because the `REDIS_PASSWORD` in `.env` is empty?

It's able to pull Redis now, which failed before. But the manifest mismatch error still persists.
